python/lsst/ip/diffim/registerImage.py

```diff
@@ -0,0 +1,192 @@
+    ('warpExposure') and sources ('warpSources') with the new Wcs.
+    """
+    ConfigClass = RegisterConfig
+
+    def run(self, inputSources, inputWcs, inputBBox, templateSources):
+        """Register (align) an input exposure to the template
+
+        The sources must have RA,Dec set, and accurate to within the
+        'matchRadius' of the configuration in order to facilitate source
+        matching.  We fit a new Wcs, but do NOT set it in the input exposure.
+
+        @param inputSources: Sources from input exposure
+        @param inputWcs: Wcs of input exposure
+        @param inputBBox: Bounding box of input exposure
+        @param templateSources: Sources from template exposure
+        @return Struct(matches: Matches between sources,
+                       wcs: Wcs for input in frame of template,
+                       )
+        """
+        matches = self.matchSources(inputSources, templateSources)
+        wcs = self.fitWcs(matches, inputWcs, inputBBox)
+        return Struct(matches=matches, wcs=wcs)
+
+    def matchSources(self, inputSources, templateSources):
+        """Match sources between the input and template
+
+        The order of the input arguments matters (because the later Wcs
+        fitting assumes a particular order).
+
+        @param inputSources: Source catalog of the input frame
+        @param templateSources: Source of the target frame
+        @return Match list
+        """
+        matches = afwTable.matchRaDec(templateSources, inputSources,
+                                      self.config.matchRadius*afwGeom.arcseconds)
+        self.log.info("Matching within %.1f arcsec: %d matches" % (self.config.matchRadius, len(matches)))
+        self.metadata.set("MATCH_NUM", len(matches))
+        if len(matches) == 0:
+            raise RuntimeError("Unable to match source catalogs")
+        return matches
+
+    def fitWcs(self, matches, inputWcs, inputBBox):
+        """Fit Wcs to matches
+
+        The fitting includes iterative sigma-clipping.
+
+        @param matches: List of matches (first is target, second is input)
+        @param inputWcs: Original input Wcs
+        @param inputBBox: Bounding box of input image
+        @return Wcs
+        """
+        copyMatches = type(matches)(matches)
+        refCoordKey = copyMatches[0].first.getTable().getCoordKey()
+        inCentroidKey = copyMatches[0].second.getTable().getCentroidKey()
+        for i in range(self.config.sipIter):
+            sipFit = makeCreateWcsWithSip(copyMatches, inputWcs, self.config.sipOrder, inputBBox)
+            self.log.debug("Registration WCS RMS iteration %d: %f pixels",
+                           i, sipFit.getScatterInPixels())
+            wcs = sipFit.getNewWcs()
+            dr = [m.first.get(refCoordKey).angularSeparation(
+                wcs.pixelToSky(m.second.get(inCentroidKey))).asArcseconds() for
+                m in copyMatches]
+            dr = numpy.array(dr)
```

Comment:
Make the list comp a generator and put it inside `np.array` to avoid the temporary.